crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2294,24 +2325,18 @@ impl<'db> TypeInferenceBuilder<'db> {
                 .unwrap_or_else(|| builtins_symbol_ty(self.db, "int").to_instance(self.db)),
 
             (Type::IntLiteral(_), Type::IntLiteral(_), ast::Operator::Div) => {
-                // TODO: division by zero error
                 builtins_symbol_ty(self.db, "float").to_instance(self.db)
             }
 
             (Type::IntLiteral(n), Type::IntLiteral(m), ast::Operator::FloorDiv) => n
                 .checked_div(m)
                 .map(Type::IntLiteral)
-                // TODO: division by zero error
-                // It should only be possible to hit this with division by zero, not an overflow.
-                // The overflow case is `i64::MIN // -1` and the negative integer is transformed
-                // into a positive integer prior to here (in the AST).
                 .unwrap_or_else(|| builtins_symbol_ty(self.db, "int").to_instance(self.db)),
 
             (Type::IntLiteral(n), Type::IntLiteral(m), ast::Operator::Mod) => n
                 .checked_rem(m)
                 .map(Type::IntLiteral)
-                // TODO division by zero error
-                .unwrap_or(Type::Todo),
+                .unwrap_or_else(|| builtins_symbol_ty(self.db, "int").to_instance(self.db)),
```

Comment:
Per discussion, division by zero (and overflows, which should never occur here) don't matter when determining the inferred type.